pkg/asset/agent/agentconfig/agenthosts.go

```diff
@@ -71,13 +70,12 @@ func (a *AgentHosts) Generate(dependencies asset.Parents) error {
 	}
 
 	if installConfig != nil && installConfig.GetBaremetalHosts() != nil {
-		icHosts := installConfig.GetBaremetalHosts()
 		if len(a.Hosts) == 0 {
-			if err := a.getInstallConfigDefaults(icHosts); err != nil {
+			if err := a.getInstallConfigDefaults(installConfig); err != nil {
 				return errors.Wrapf(err, "invalid host definition in %s", agentAsset.InstallConfigFilename)
 			}
 		}
-		if len(a.Hosts) > 0 && len(icHosts) > 0 {
+		if len(a.Hosts) > 0 && len(installConfig.GetBaremetalHosts()) > 0 {
```

Comment:
(very) nit: len check could be moved at line 73, substituting nil check and saving one

Reply:
Another minor point, the warning message for ignoring eventual hosts defined in agent-config could be printed before `getInstallConfigDefaults`

Reply:
Fixed, and add comment clarifying host source.